### PR TITLE
Datadog - kinesis setup

### PR DESCRIPTION
**Pull request by dudymas**
## what
* provide one way of setting up kinesis with datadog

## why
* reduces latency in metric collection
* solution from datadog uses cloudformation and is not as easy to configure/extend

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -4,6 +4,30 @@ variable "datadog_aws_account_id" {
   default     = "464622532012"
 }
 
+variable "enable_kinesis" {
```
Comment (aknysh) [inline variables.tf:7]: looks like this var is not used

**Review by aknysh — COMMENTED** [review]
please see comments

File: variables.tf (hunk 1 above)
Comment (dudymas) [reply variables.tf:7]: good call. I think this should be fixed now. I'm using it as a 'count' on the kinesis module itself

File: kinesis/firehose_iam.tf [hunk 2]
```diff
@@ -0,0 +1,43 @@
+
```
Comment (nitrocode) [inline kinesis/firehose_iam.tf:1]:
```suggestion
```

File: kinesis/firehose_iam.tf [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+resource "aws_iam_role" "datadog_firehose" {
+  name               = "${module.this.id}-firehose"
+  assume_role_policy = data.aws_iam_policy_document.firehose_sts.json
+
+  inline_policy {
+    name   = "${module.this.id}-backup"
+    policy = data.aws_iam_policy_document.datadog_firehose_s3_backup.json
+  }
+}
```
Comment (nitrocode) [inline kinesis/firehose_iam.tf:43]:
```suggestion

  tags = module.this.tags
}
```

File: kinesis/main.tf [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# significant inspiration from the following sources:
+# https://github.com/magnetikonline/terraform-aws-datadog-metric-stream
+
+data "aws_caller_identity" "current" {}
```
Comment (nitrocode) [inline kinesis/main.tf:4]: Missing count

File: kinesis/main.tf [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+data "aws_ssm_parameter" "datadog_api_key" {
+  name = var.datadog_api_key_ssm_parameter_name
```
Comment (nitrocode) [inline kinesis/main.tf:7]: Missing count

File: kinesis/main.tf [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+locals {
+  backup_bucket_arn = var.backup_bucket_arn
+  datadog_api_key   = data.aws_ssm_parameter.datadog_api_key.value
+}
+
+## Kinesis Firehose
+resource "aws_kinesis_firehose_delivery_stream" "datadog" {
+  name        = "datadog"
```
Comment (nitrocode) [inline kinesis/main.tf:17]: Missing count

File: kinesis/main.tf [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  server_side_encryption {
+    enabled = false
+  }
+}
+
+## CloudWatch metric stream
+resource "aws_cloudwatch_metric_stream" "datadog" {
+  name          = module.this.id
```
Comment (nitrocode) [inline kinesis/main.tf:62]: Missing count

File: variables.tf (hunk 1 above)
Comment (nitrocode) [reply variables.tf:7]: This var should also be

```suggestion
variable "kinesis_enabled" {
```

File: main.tf [hunk 8]
```diff
@@ -25,6 +25,16 @@ resource "datadog_integration_aws" "integration" {
   account_specific_namespace_rules = var.account_specific_namespace_rules
 }
 
+module "kinesis" {
+  count   = var.kinesis_enabled ? 1 : 0
```
Comment (nitrocode) [inline main.tf:29]: The module should respect the enabled flag on it which is passed by the context but also by `var.kinesis_enabled`

```suggestion
  enabled = local.enabled && var.kinesis_enabled ? 1 : 0
```

**Comment by nitrocode** [comment]
Please do the following

* tag all respective resources
* disable resources if enabled is false, usually we use `locals { enabled = module.this.enabled }` and then ref `local.enabled ? 1 : 0` for each `count` on each resource.

**Pull request closed without merging**